inc/umock_c/umock_c_internal.h

```diff
@@ -23,6 +23,12 @@
 #include "umock_c/umockstring.h"
 #include "umock_c/umockautoignoreargs.h"
 
+// For gcc build that causes a false negative
+#ifndef _MSC_VER
+#pragma GCC diagnostic ignored "-Wunused-variable"
```

Comment:
>pragma [](http://example.com/codeflow?start=1&length=6)

These Pragmas once included through this header - they will eliminate any unused-variable warning from EVERYWHERE.

Is there any way to scope these warnings to ONLY the variables which are owned by .... umock and leave all the others untouched?

Reply:
Let's try and make an effort and scope the ignore only to the variable that are involuntarily introduced by umock in the user space.

Reply:
hmmm... looking on the left side, seems like this change would need to be backported to logging/testrunnerswitch/ctest/macro_utils., Cough.

Reply:
Signing off because I know you're going to do it :)

Reply:
Yes, I will back port all the other repos with this change.  Adding this Pragma didn't see to have global scope, because other functions gave me this error too.  I'll dig into the documentation further to be sure, but I think this is scoped